File: app/src/main/java/se/appshack/android/refactoring/main/data/model/PokemonDetailsResponse.java
package se.appshack.android.refactoring.main.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PokemonDetailsResponse {

    @SerializedName("id")
    public int id;

    @SerializedName("name")
    public String name;

    @SerializedName("height")
    public int height;

    @SerializedName("weight")
    public int weight;

    @SerializedName("species")
    public NamedResponseModel species;

    @SerializedName("types")
    public List<PokemonTypeModel> types;

    @SerializedName("sprites")
    public PokemonSpritesModel sprites;
}

File: app/src/main/java/se/appshack/android/refactoring/main/data/model/PokemonDetailsResponse.kt
package se.appshack.android.refactoring.main.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class PokemonDetailsResponse (

    @SerializedName("id")
    var id: Int = 0,

    @SerializedName("name")
    var name: String? = null,

    @SerializedName("height")
    var height: Int = 0,

    @SerializedName("weight")
    var weight: Int = 0,

    @SerializedName("species")
    var species: NamedResponseModel? = null,

    @SerializedName("types")
    var types: List<PokemonTypeModel>? = null,

    @SerializedName("sprites")
    var sprites: PokemonSpritesModel? = null

): Parcelable

